### PR TITLE
Add CompetionDAO interface

### LINKED ISSUE #5: Configure maven dependencies

### LINKED ISSUE #12: Create entity class Competition

### PR DESCRIPTION
**Pull request by sithcom255**
-add the implementation for it also
-add basic spring dependencies
-basic competion entity

### CODE REVIEW

File: src/main/java/cz/fi/muni/pa165/esports/dao/CompetitionDao.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package cz.fi.muni.pa165.esports.dao;
+
+import cz.fi.muni.pa165.esports.entity.Competition;
+
+import java.util.List;
+
+/**
+ * @author jan gavlik
+ */
+:
```
Comment (kaprijela) [inline src/main/java/cz/fi/muni/pa165/esports/dao/CompetitionDao.java]: Please remove this extra colon `:`, the build fails when it's there :)
```suggestion

```

**Review by kaprijela — CHANGES_REQUESTED** [review]
There is an extra colon in `CompetitionDao.java` on line 10, please remove it so the build can pass (doing that in another commit is fine, or you can commit the suggested change straight from this pull request page, see below).

Other than that I think this is okay, so I'll merge it to main when it's done :+1: 

Also I guess we don't need it now, but maybe it would make sense to have methods like `findByLocation`, `findByGame` for competitions? What do you think?

**Comment by kaprijela** [comment]
Also, I'm thinking -- should we also implement `hashCode` and `equals` in our classes right away? @sithcom255 @elenuki33 @RadovanTomik

File: src/main/java/cz/fi/muni/pa165/esports/dao/CompetitionDao.java (hunk 1 above)
Comment (sithcom255) [reply src/main/java/cz/fi/muni/pa165/esports/dao/CompetitionDao.java]: aaa thank you,

**Review by kaprijela — APPROVED** [review]
I think everything is all right now :+1:

**Pull request merged**